ntbs-service/Services/AuthorizationService.cs

```diff
@@ -37,78 +36,99 @@ public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Aler
+            if (_filter == null)
+            {
+                _filter = await GetUserPermissionsFilterAsync(user);
+            }
+
+            if (_filter.FilterByTBService)
+            {
+                return _filter.IncludedTBServiceCodes.Contains(notificationBannerModel.TbServiceCode);
+            }
+
+            if (_filter.FilterByPHEC)
+            {
+                var allowedCodes = _filter.IncludedPHECCodes;
+                return allowedCodes.Contains(notificationBannerModel.TbServicePHECCode) ||
+                       allowedCodes.Contains(notificationBannerModel.LocationPHECCode);
+            }
+            return true;
+            
         }
 
         public async Task<PermissionLevel> GetPermissionLevelForNotificationAsync(ClaimsPrincipal user,
-            Notification notification, NotificationGroup group = null)
+            Notification notification)
         {
-            if (await CanEditNotificationAsync(user, notification))
+            if (_filter == null)
+            {
+                _filter = await GetUserPermissionsFilterAsync(user);
+            }
+            
+            if (UserHasDirectRelationToNotification(notification))
             {
                 return PermissionLevel.Edit;
             }
 
-            if (await CanEditLinkedNotificationAsync(user, group))
+            if (UserHasDirectRelationToLinkedNotification(notification?.Group?.Notifications))
             {
                 return PermissionLevel.ReadOnly;
             }
 
             return PermissionLevel.None;
         }
-
-        public async Task<bool> CanEditLinkedNotificationAsync(ClaimsPrincipal user, NotificationGroup group)
+        
+        private bool UserHasDirectRelationToLinkedNotification(IEnumerable<Notification> linkedNotifications)
         {
-            return group?.Notifications.Select(n => CanEditNotificationAsync(user, n).Result).Any(x => x == true) ??
-                   false;
+            return linkedNotifications.Select(UserHasDirectRelationToNotification).Any(x => x == true);
         }
 
-        public async Task<bool> CanEditNotificationAsync(ClaimsPrincipal user, Notification notification)
+        private bool UserHasDirectRelationToNotification(Notification notification)
         {
-            var tbServiceCode = notification.Episode.TBServiceCode;
-            var tbServicePhecCode = notification.Episode.TBService?.PHECCode;
-            var locationPhecCode = notification.PatientDetails.PostcodeLookup?.LocalAuthority?.LocalAuthorityToPHEC?.PHECCode;
-            return await AuthorizeUserAccessForTbServiceAndPhecCodes(user, tbServiceCode, locationPhecCode, tbServicePhecCode);
+            return UserBelongsToTbServiceOfNotification(notification) || UserBelongsToTreatmentPhecOfNotification(notification) ||
+                UserBelongsToResidencePhecOfNotification(notification);

```

Comment:
Welp! I think we're forgetting an important condition here - User is in the national team!

Reply:
github seems to have accepted my commit of 'account for national team users' but not actually shown the changes, I've put it as a global kind of national team gets edit access check.

The code should read if (UserHasDirectRelationToNotification(notification) || _filter.Type == UserType.NationalTeam